FRS documentation with TokenProvider and Azure function (#7052)
microsoft/FluidFramework: docs/content/docs/deployment/azure-frs.md

### Token providers

The [FrsAzFunctionTokenProvider]({{< relref "https://github.com/microsoft/FluidFramework/blob/main/experimental/framework/frs-client/src/FrsAzFunctionTokenProvider.ts" >}}) is an implementation of `ITokenProvider` which ensures your tenant key secret is not exposed in your client-side bundle code. The `FrsAzFunctionTokenProvider` takes in your Azure Function URL appended by `/api/GetFrsToken` along with the current user object. Later on, it makes an axios `GET` request call to your Azure function by passing in the tenantID, documentId and userID/userName as optional parameters.

```javascript
const config = {
    tenantId: "myTenantId",

>     tokenProvider: new FrsAzFunctionTokenProvider("AZURE-FUNCTION-URL"+"/api/GetFrsToken", { userId:

Reviewer: ```suggestion
    tokenProvider: new FrsAzFunctionTokenProvider("myAzureFunctionUrl"+"/api/GetFrsToken", { userId:
```

Reviewer: just to be consistent with the other strings here 😄

    "UserId", userName: "Test User"}),
    orderer: "https://myOrdererUrl",
    storage: "https://myStorageUrl",
};

const client = new FrsClient(config);
```

> The user object can also hold optional additional user details such as the gender, address, email, etc.

Reviewer: ```suggestion
The user object can also hold optional additional user details such as the gender, address, email, etc. For example:
```

```javascript
cont userDetails = {
  email: "xyz@outlook.com",
  address: "Redmond",
};

> 

Reviewer: Let's also specify that the developer can add in any other fields here for this object that they would like to see returned in the `FrsMember` values from the audience in container services. I think a code sample showing the same additionalDetails field in the example above being parsed from audience would be helpful to show the entire loop

const audienceDiv = document.createElement("div");

const onAudienceChanged = () => {
        const members = audience.getMembers();
        const self = audience.getMyself();
        const memberStrings: string[] = [];
        const useFrs = process.env.FLUID_CLIENT === "frs";

        members.forEach((member: FrsMember<ICustomUserDetails>) => {
            if (member.userId !== self?.userId) {
                if (useFrs) {
                    const memberString = `${member.userName}: {Email: ${member.additionalDetails?.email},
                        Address: ${member.additionalDetails?.address}}`;
                    memberStrings.push(memberString);
                } else {
                    memberStrings.push(member.userName);
                }
            }
        });
        audienceDiv.innerHTML = `
            Current User: ${self?.userName} <br />
            Other Users: ${memberStrings.join(", ")}
        `;
    };

    onAudienceChanged();
    audience.on("membersChanged", onAudienceChanged);
```

> `audience` provides two functions that will return `FrsMember` objects that have a user ID and user name:

Reviewer: I believe we also need to update the sample `FrsMember` value below to include the additionalDetails now

Reviewer: Also, we should specifically state that the values in those fields match up to the values provided in the token

Reviewer 2: Yep! Sorry missed that

      "id": "0e662aca-9d7d-4ff0-8faf-9f8672b70f15",
      "mode": "write"
    }
  ],
  "additionalDetails": {
      "email": "xyz@outlook.com",
      "address": "Redmond",
  }
}
```

> Alongside the user ID, name and addiitonalDetails, `FrsMember` objects also hold an array of `connections`. If the user is logged into the session with only one client, `connections` will only have one value in it with the ID of the client and if is in read/write mode. However, if the same user is logged in from multiple clients (i.e. they are logged in from different devices or have multiple browser tabs open with the same container), `connections` here will hold multiple values for each client. In the example data above, we can see that a user with name "Test User" and ID "0e662aca-9d7d-4ff0-8faf-9f8672b70f15" currently has the container open from two different clients. The values in the `additionalDetails` field match up to the values provided in the `FrsAzFunctionTokenProvider` token generation.

Reviewer: ```suggestion
Alongside the user ID, name and additional details, `FrsMember` objects also hold an array of `connections`. If the user is logged into the session with only one client, `connections` will only have one value in it with the ID of the client and if is in read/write mode. However, if the same user is logged in from multiple clients (i.e. they are logged in from different devices or have multiple browser tabs open with the same container), `connections` here will hold multiple values for each client. In the example data above, we can see that a user with name "Test User" and ID "0e662aca-9d7d-4ff0-8faf-9f8672b70f15" currently has the container open from two different clients. The values in the `additionalDetails` field match up to the values provided in the `FrsAzFunctionTokenProvider` token generation.
```
